webapp/runtime/src/main/java/org/wildfly/swarm/topology/webapp/runtime/TopologyProxyService.java

```diff
@@ -97,8 +98,8 @@ public void onChange(Topology topology) {
     }
 
     private void updateProxyHosts(String serviceName, List<Topology.Entry> entries) {
-        ProxyHandler proxyHandler = proxyHandlerMap.get(serviceName).getValue();
-        LoadBalancingProxyClient proxyClient = (LoadBalancingProxyClient) proxyHandler.getProxyClient();
+        HttpHandler proxyHandler = proxyHandlerMap.get(serviceName).getValue();
```

Comment:
I don't mind if it's nice. I am just surprised that it works. If I understand you correctly this was the root cause:

```
Cannot cast org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler to io.undertow.server.handlers.proxy.ProxyHandler
```

But GlobalRequestControllerHandler is just an HttpHandler. So if it passes the type cast, does that mean it will not be injected anymore?